docs/license-handling.asciidoc

```diff
@@ -0,0 +1,63 @@
+[id="{p}-licensing"]
+== Managing licenses in ECK
+
+When you install the default distribution of ECK, you receive a basic license. Any Elastic stack application you manage through
+ECK will also be basic licensed. For the full list of free features that are included in the basic license, see: https://www.elastic.co/subscriptions
+
+[float]
+=== Starting a trial
+If you want to try the features included in the Enterprise subscription, you can start a 30-day trial. To start a trial create
+a Kubernetes secret as shown below:
+
+[source,yaml]
+----
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: any-name-you-like
+  namespace: elastic-system
+  labels:
+    licencse.k8s.elastic.co/type: enterprise-trial
+  annotations:
+    elastic.co/eula: accepted <1>
+EOF
+----
+
+<1> By setting this annotation to `accepted` you are expressing that you have accepted the Elastic EULA which can be found at https://www.elastic.co/eula
+
+NOTE: You can initiate a trial license only if a trial has not previously been activated for the current major product version of your ECK installation.
+
+At the end of the trial period, the platinum and enterprise features operate in a link:https://www.elastic.co/guide/en/elastic-stack-overview/7.3/license-expiration.html[degraded mode]. You can revert to a basic license, extend the trial, or purchase an enterprise subscription.
+
+[float]
+=== Adding a license
+If you have a valid enterprise subscription you will receive a license as a JSON file.
+To add the license to your ECK installation, create a Kubernetes secret of the following form:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    license.k8s.elastic.co/type: enterprise <1>
+  name: your-choice
+type: Opaque
+data:
+  license: "your license JSON goes here"  <2>
```

Comment:
This should be a base64 encoded string.
An other option is to use the `stringData` field:

```yaml
----
apiVersion: v1
kind: Secret
metadata:
  labels:
    license.k8s.elastic.co/type: enterprise <1>
  name: your-choice
type: Opaque
stringData:
  license: "this does not need to be a base64 encoded license"
```

Reply:
Maybe
```
license: "JSON license in base64 format"
```

would be good enough since the next paragraph explains how to set it up anyway.